### PR TITLE
Proxy Library RPC Generation

### PR DESCRIPTION
**Pull request by joeljfischer**
This proposal adds automatic RPC generation for iOS and Java from the XML spec via a python script.

### CODE REVIEW

**Comment by joeljfischer** [comment]
@theresalech This is ready for review

File: proposals/nnnn-proxy-rpc-generation.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+* Proposal: [SDL-NNNN](NNNN-proxy-rpc-generation.md)
+* Author: [SDL Developer](https://github.com/joeljfischer)
+* Status: **Awaiting review**
+* Impacted Platforms: [iOS / Java Suite]
+
+## Introduction
+This proposal adds automatic RPC generation for iOS and Java from the XML spec via a python script.
+
+## Motivation
+Adding RPCs is currently difficult. Every RPC to be added requires hand-writing each parameter, then review, and then unit testing. This makes large RPC changes very difficult, and at times, error-prone. Beyond that, alterations to the RPC classes require an enormous amount of work due to the number of RPCs that already exist – in addition to the problems of code duplication. This does not happen often, but it does happen.
```
Comment (theresalech) [inline proposals/nnnn-proxy-rpc-generation.md]:
```suggestion
Adding RPCs is currently difficult. Every RPC to be added requires handwriting each parameter, then review, and then unit testing. This makes large RPC changes very difficult, and at times, error-prone. Beyond that, alterations to the RPC classes require an enormous amount of work due to the number of RPCs that already exist – in addition to the problems of code duplication. This does not happen often, but it does happen.
```

File: proposals/nnnn-proxy-rpc-generation.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
+# Proxy Library RPC Generation
+
+* Proposal: [SDL-NNNN](NNNN-proxy-rpc-generation.md)
+* Author: [SDL Developer](https://github.com/joeljfischer)
```
[the remark this answers is not in the document]
Comment (joeljfischer) [reply proposals/nnnn-proxy-rpc-generation.md]: https://media.giphy.com/media/NVoJyy7okYb72/giphy.gif

**Comment by joeljfischer** [comment]
Fixed!

**Pull request merged**